### PR TITLE
User can search for song

### PR DESCRIPTION
**Pull request by KyunginNa**
[API: User can search for a song to post](https://www.pivotaltracker.com/story/show/176623790)

## User Story
```
As an API,            
In order to provide the client with search results,         
I want to retrieve data from the Spotify API.
```

## Changes proposed in this pull request:
* Creating a test if the user can search for a song to post
* Initializing and configuring RSpotify gem with credentials
* Generating Tracks controller and creating an index action
* Adding a route for the index action

## What I have learned working on this feature:
* How to fix the format of the response fetched from RSpotify gem
* How to implement search function using index action

## Packages/Gems added
* RSpotify

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,4 +22,7 @@
 /config/master.key
 
 # Ignore Coverage
-/coverage
\ No newline at end of file
+/coverage
+
+# Ignore Spotify credentials
+/config/initializers/spotify_setup.rb
```
Comment (tochman) [inline .gitignore]: That is not how you hide credentials from version control. Please use the encrypted credentials method in Rails and reinstate this file to VCS

File: spec/requests/api/user_can_search_for_a_song_to_post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+RSpec.describe 'GET /api/tracks', type: :request do
+  describe 'successfully get a list of songs' do
+    before do
+      get '/api/tracks', params: { q: 'All I Want for Christmas Is You' }
```
Comment (tochman) [inline spec/requests/api/user_can_search_for_a_song_to_post_spec.rb:4]: You want to set up WebMock so you avoid making the call in test env.

Comment (sealfury) [reply spec/requests/api/user_can_search_for_a_song_to_post_spec.rb:4]: This seems like as good a time as any to check my account details and let you know how many calls we can make for when we start with dry runs

Comment (rithreaksa) [reply spec/requests/api/user_can_search_for_a_song_to_post_spec.rb:4]: > This seems like as good a time as any to check my account details and let you know how many calls we can make for when we start with dry runs

What do you mean? 😄

Comment (rithreaksa) [reply spec/requests/api/user_can_search_for_a_song_to_post_spec.rb:4]: > You want to set up WebMock so you avoid making the call in test env.

We fixed that. Thanks.

File: .gitignore (hunk 1 above)
Comment (KyunginNa) [reply .gitignore]: This one is fixed now!

File: spec/requests/api/user_can_search_for_a_song_to_post_spec.rb (hunk 2 above)
Comment (KyunginNa) [reply spec/requests/api/user_can_search_for_a_song_to_post_spec.rb:4]: @tochman We set up Webmock👍

**Pull request merged**